UnityProject/Assets/Scripts/Items/Weapons/Gun.cs

```diff
@@ -729,15 +729,29 @@ private bool TryReload(GameObject ammo)
 				SoundManager.PlayNetworkedAtPos(FiringSoundA, shooter.transform.position);
 			}
 
-			RPCShowMuzzleFlashAndRecoil(shooter.GetComponent<NetworkIdentity>(), finalDirection);
+			var Identity = shooter.GetComponent<NetworkIdentity>();
+			if (Identity.OrNull()?.connectionToClient != null)
+			{
+				if (isServer && shooter == PlayerManager.LocalPlayer)
+				{
+					Camera2DFollow.followControl.Recoil(-finalDirection, CameraRecoilConfig);
+				}
 
+				RPCShowMuzzleFlash(shooter.GetComponent<NetworkIdentity>());
```

Comment:
Muzzle flash should be shown even if `connectionToClient` is `null`, as it is shown to everyone.